src/Tests/dotnet-run.Tests/GivenDotnetRunRunsCsProj.cs

```diff
@@ -208,6 +211,54 @@ public void ItWarnsWhenShortFormOfProjectArgumentIsUsed()
                          .And.HaveStdOutContaining(LocalizableStrings.RunCommandProjectAbbreviationDeprecated);
         }
 
+        [Fact]
+        public void ItErrorsWhenMultipleProjectsAreSpecified()
+        {
+            new DotnetCommand(Log, "run")
+                .Execute($"-p", "project1", "-p", "project2")
+                .Should()
+                .Fail()
+                .And
+                .HaveStdErrContaining(LocalizableStrings.OnlyOneProjectAllowed);
+        }
+
+        [Theory]
+        [InlineData("--property:prop1=true", "Prop1: true, Prop2: ")]
+        [InlineData("--property prop1=true", "Prop1: true, Prop2: ")]
+        [InlineData("-p:prop1=true", "Prop1: true, Prop2: ")]
+        [InlineData("-p prop1=true", "Prop1: true, Prop2: ")]
+        [InlineData("-p prop1=true -p prop2=false", "Prop1: true, Prop2: false")]
+        public void ItAcceptsPropertyOption(string optionArgs, string expectedPropertyOutput)
```

Comment:
functional tests start from _dotnet_ are slow and harder to maintain. It also won't help you create abstraction. It would be better to unit test in parser layer. And just have 1 functional test